snuba/query/processors/uuid_column_processor.py

```diff
@@ -0,0 +1,112 @@
+from typing import Set
+import uuid
+
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.conditions import (
+    binary_condition,
+    is_condition,
+    is_in_condition,
+    is_not_in_condition,
+)
+from snuba.query.logical import Query
+from snuba.query.processors import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class UUIDColumnProcessor(QueryProcessor):
+    """
+    If a condition is being performed on a column that stores UUIDs (as defined in the constructor)
+    then change the condition to use a proper UUID instead of a string.
+    """
+
+    def __init__(self, uuid_columns: Set[str]) -> None:
+        self.__uuid_columns = uuid_columns
+
+    def is_uuid_column(self, exp: Expression) -> bool:
+        return isinstance(exp, Column) and exp.column_name in self.__uuid_columns
+
+    def is_formatted_uuid(self, exp: Expression) -> bool:
+        return (
+            isinstance(exp, FunctionCall)
+            and exp.function_name == "replaceAll"
+            and isinstance(exp.parameters[0], FunctionCall)
+            and exp.parameters[0].function_name == "toString"
+            and isinstance(exp.parameters[0].parameters[0], Column)
+            and exp.parameters[0].parameters[0].column_name in self.__uuid_columns
+        )
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        def process_condition(exp: Expression) -> Expression:
+            if is_condition(exp):
+                assert isinstance(exp, FunctionCall)
```

Comment:
Same for this logic, I think using a pattern to find your expression would make it a lot easier.
I think you are looking for a pattern like this:
```
FunctionCall(None, 
    Param("operator", Or([your operators])), 
    (FunctionCall(None, String("ReplaceAll"), (
         FucntionCall(None, String("toString), (
           Param("uuid_column", Column(None, None, Or([your columns])),
           Param("uuid_literals", AnyExpression())
        )
    )
```

Then you have the expression you are trying to transform matched and the operator, and operands populated.     

Should you need to add more Pattern matchers in matchers.py feel free to add them.